### PR TITLE
Fixed #26616 -- Improves error message when AppConfig.name is invalid

### PR DESCRIPTION
**Pull request by inondle**

### CODE REVIEW

**Comment by inondle** [comment]
Does anyone know where I could put a test for this particular change? I've tried looking for some keywords and they didn't turn anything up. I'll also look through the docs and see where I can add to them.

**Comment by inondle** [comment]
I updated the test so that it expects an ImproperlyConfigured error rather than ImportError. Still not sure how/where to change docs though.

File: django/apps/config.py [hunk 1]
```diff
@@ -139,7 +139,11 @@ def create(cls, entry):
                 "'%s' must supply a name attribute." % entry)
 
         # Ensure app_name points to a valid module.
-        app_module = import_module(app_name)
+        try:
+            app_module = import_module(app_name)
+        except ImportError:
+            raise ImproperlyConfigured(
+                "Cannot import '%s', check AppConfig.name is correctly defined.")
```
Comment (berkerpeksag) [inline django/apps/config.py]: There is a `%s` in the exception message, but you forgot to pass `app_name`.

A minor code style suggestion:

```py
raise ImproperlyConfigured(
    "Cannot import '%s', check AppConfig.name is correctly defined." % app_name
)
```

Comment (timgraham) [inline django/apps/config.py]: To that end, you should use `assertRaisesMessage` in the test to catch that mistake.

**Comment by inondle** [comment]
@berkerpeksag 
@timgraham 

Fixed those typos and updated the test

File: tests/apps/tests.py [hunk 2]
```diff
@@ -78,7 +78,8 @@ def test_no_such_app(self):
         with self.assertRaises(ImportError):
             with self.settings(INSTALLED_APPS=['there is no such app']):
                 pass
-        with self.assertRaises(ImportError):
+        msg = "Cannot import 'there is no such app', check AppConfig.name is correctly defined."
```
Comment (timgraham) [inline tests/apps/tests.py]: Could this message be improved with the actual AppConfig class name and path?

Comment (inondle) [inline tests/apps/tests.py]: @timgraham 
I could change it to something like:

`"Cannot import '%s' check '%s.%s.name' is correctly defined." % (app_name, mod_path, cls_name)`

which would render to:

`"Cannot import 'there is no such app' check 'apps.apps.NoSuchApp.name' is correctly defined."`

Look good?

**Comment by timgraham** [comment]
Yes, that's what I had in mind. The comma in the error message could be a period and then "Check that apps.apps.NoSuchApp.name is correct."

**Comment by timgraham** [comment]
merged in 080dd74e016fbc99d3aaecd36ef932424042b768, thanks!

**Pull request closed without merging**